homeassistant/components/sensor/gtfs.py

```diff
@@ -48,73 +50,143 @@
     vol.Required(CONF_DATA): cv.string,
     vol.Optional(CONF_NAME): cv.string,
     vol.Optional(CONF_OFFSET, default=0): cv.time_period,
+    vol.Optional(CONF_TOMORROW, default=False): cv.boolean,
 })
 
 
-def get_next_departure(sched, start_station_id, end_station_id, offset):
+def get_next_departure(sched, start_station_id, end_station_id, offset,
+                       include_tomorrow=False):
     """Get the next departure for the given schedule."""
     origin_station = sched.stops_by_id(start_station_id)[0]
     destination_station = sched.stops_by_id(end_station_id)[0]
 
     now = datetime.datetime.now() + offset
-    day_name = now.strftime('%A').lower()
-    now_str = now.strftime('%H:%M:%S')
-    today = now.strftime('%Y-%m-%d')
+    now_date = now.strftime(DATE_FORMAT)
+    yesterday = now - datetime.timedelta(days=1)
+    yesterday_date = yesterday.strftime(DATE_FORMAT)
+    tomorrow = now + datetime.timedelta(days=1)
+    tomorrow_date = tomorrow.strftime(DATE_FORMAT)
 
     from sqlalchemy.sql import text
 
-    sql_query = text("""
```

Comment:
Previously, the query would only retrieve the 1 departure today (based on date + day of the week) after the set time. Because there can be departures at 1 in the morning that belong to yesterday, and departures after midnight tomorrow that belong to today, we are missing out on those results. And since yesterday and tomorrow could potentially have different schedules from today (think weekdays vs weekends) we cannot just map today's after-midnight departures as this morning's.